### PR TITLE
adjust wording on rpk group delete

### PR DESCRIPTION
**Pull request by Deflaimun**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2444
Review deadline: June 20

## Page previews

https://deploy-preview-561--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-group/rpk-group-delete/

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc [hunk 1]
```diff
@@ -1,23 +1,13 @@
 = rpk group delete
 
-Delete groups from brokers.
+Delete Consumer Groups
```
Comment (JakeSCahill) [inline modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]:
```suggestion
Delete consumer groups
```

File: modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc [hunk 2]
```diff
@@ -1,23 +1,13 @@
...
-empty for offset.retention.minutes. Redpanda does not currently support that
-configuration (see #2904), meaning offsets for empty groups expire only when
-they are explicitly deleted.
+Group offset deletion can happen through:
 
-You may want to delete groups to clean up offsets sooner than when they
-automatically are cleaned up, such as when you create temporary groups for
-quick investigation or testing. This command helps you do that.
+* Periodic Offset Expiration: Offsets expire automatically when the group has been empty for a set duration. See xref:reference:properties/cluster-properties.adoc#group_offset_retention_sec[Cluster properties - `group_offset_retention_sec`]
+* Kafka `OffsetDelete` API: Offsets can be explicitly deleted using the Kafka `OffsetDelete` API.
```
Comment (kbatuigas) [inline modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]: I wasn't able to find the official Kafka or Confluent docs for this. Is this the equivalent: https://docs.redpanda.com/current/reference/rpk/rpk-group/rpk-group-offset-delete/ and should we just point readers to that? cc @twmb or @r-vasquez ?

File: modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc [hunk 3]
```diff
@@ -1,23 +1,13 @@
 = rpk group delete
 
-Delete groups from brokers.
+Delete consumer groups
 
-Older versions of the Kafka protocol included a retention_millis field in
-offset commit requests. Group commits persisted for this retention and then
-eventually expired. Once all commits for a group expired, the group would be
-considered deleted.
+Redpanda supports deleting consumer groups manually through `rpk group delete` to manage offsets proactively. This is useful for cleaning up offsets sooner than the automatic cleanup, such as when creating temporary groups for investigations or testing.
```
Comment (kbatuigas) [inline modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]:
```suggestion
Delete consumer groups manually through `rpk group delete`. This allows you to proactively manage offsets, for example, when you've created temporary groups for quick investigation or testing and you want to clear offsets sooner than the automatic cleanup.
```

**Review by kbatuigas — COMMENTED** [review]
Clarifying comment and a suggestion (would be helpful to check with SME as well)

File: modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc (hunk 2 above)
Comment (twmb) [reply modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]: It is equivalent, probably better to redirect

Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]: Added a link to the rpk command at the end of the sentence.

File: modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc [hunk 4]
```diff
@@ -1,23 +1,13 @@
 = rpk group delete
 
-Delete groups from brokers.
+Delete consumer groups
 
-Older versions of the Kafka protocol included a retention_millis field in
-offset commit requests. Group commits persisted for this retention and then
-eventually expired. Once all commits for a group expired, the group would be
-considered deleted.
+Delete consumer groups manually through `rpk group delete`. This allows you to proactively manage offsets, for example, when you've created temporary groups for quick investigation or testing and you want to clear offsets sooner than the automatic cleanup.
```
Comment (r-vasquez) [inline modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]:
```suggestion
This command lets you explicitly delete consumer groups. 

This allows you to proactively manage offsets, for example, when you've created temporary groups for quick investigation or testing and you want to clear offsets sooner than the automatic cleanup.
```

File: modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc [hunk 5]
```diff
@@ -1,23 +1,13 @@
...
 
-The retention field was removed because it proved problematic for infrequently
-committing consumers: the offsets could be expired for a group that was still
-active. If clients use new enough versions of OffsetCommit (versions that have
-removed the retention field), brokers expire offsets only when the group is
-empty for offset.retention.minutes. Redpanda does not currently support that
-configuration (see #2904), meaning offsets for empty groups expire only when
-they are explicitly deleted.
+Group offset deletion can happen through:
```
Comment (r-vasquez) [inline modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]:
```suggestion
Consumer groups are automatically deleted when the last committed offset expires. Group offset deletion can happen through:
```

**Review by r-vasquez — COMMENTED** [review]
Just a small comment to help clarify to new users why we are mentioning offsets in a consumer group deletion command.

File: modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc (hunk 4 above)
Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-group/rpk-group-delete.adoc]:
```suggestion
Delete consumer groups explicitly through `rpk group delete`.

This allows you to proactively manage offsets, for example, when you've created temporary groups for quick investigation or testing and you want to clear offsets sooner than the automatic cleanup.
```

**Review by r-vasquez — APPROVED** [review]
Thanks! LGTM

**Review by kbatuigas — APPROVED** [review]
lgtm

**Pull request merged**